crypto-sodium/lib/Crypto/Encrypt/Public.hs

```diff
@@ -111,3 +119,16 @@ decrypt
   -> ctBytes -- ^ Encrypted message (cyphertext)
   -> Maybe ptBytes
 decrypt = NaCl.Box.open
+
+
+-- | Generate a new 'SecretKey' together with its 'PublicKey' from a given seed.
+keypairFromSeed
+  :: ByteArrayAccess seed
+  => seed
+  -> IO (PublicKey ByteString, SecretKey ScrubbedBytes)
```

Comment:
This function is deterministic in pure, so there should be a version without `IO`!

Reply:
Btw I'm not sure if I should use `unsafePerformIO` or `unsafeDupablePerformIO` here, as I remember you had some doubts about that in some other issue. However, the docs suggest that the problems mostly arise from partially run commands interrupted in the middle, and I (tentatively) don't think generating key pair from seed specifically will be problematic if interrupted.
Considering the latter is supposed to be more efficient, I went with it. :shrug:

Reply:
> I remember you had some doubts about that in some other issue.

It was not some other issue, it was exactly this one :). And I am still not convinced that it is safe (see #18).